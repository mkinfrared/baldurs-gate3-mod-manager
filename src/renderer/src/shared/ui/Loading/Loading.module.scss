.Loading {
  --color: 255, 255, 255;
  --background: rgb(var(--color) 0.05);

  @media (prefers-color-scheme: light) {
    --color: 0, 0, 0;
  }

  align-items: center;
  background-color: var(--background);
  display: flex;
  justify-content: center;
  left: 0;
  max-height: 100vh;
  position: absolute;
  top: 0;
  width: 100%;

  .spinner {
    aspect-ratio: initial;
    height: 50px;
    width: 50px;
  }
}
