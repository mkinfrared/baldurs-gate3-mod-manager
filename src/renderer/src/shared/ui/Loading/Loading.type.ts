import { SpinnerProps } from "../Spinner";

export type LoadingProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  type?: SpinnerProps["type"];
};
