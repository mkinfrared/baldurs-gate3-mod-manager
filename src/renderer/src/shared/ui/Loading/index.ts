export { default } from "./Loading";

export * from "./Loading.type";
