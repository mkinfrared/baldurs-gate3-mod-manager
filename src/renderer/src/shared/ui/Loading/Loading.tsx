import { useLayoutEffect, useRef } from "react";

import { classNames } from "@renderer/shared/lib/helpers";

import Spinner from "../Spinner";

import css from "./Loading.module.scss";
import { LoadingProps } from "./Loading.type";

const Loading = ({ className, type }: LoadingProps) => {
  const ref = useRef<HTMLDivElement>(null);

  useLayoutEffect(() => {
    const { current } = ref;

    if (current?.parentElement) {
      const height = current.parentElement.clientHeight;
      const { position } = getComputedStyle(current.parentElement);

      if (position === "static") {
        current.parentElement.style.position = "relative";
      }

      current.parentElement.style.height = `${height}px`;

      current.parentElement.style.overflow = "hidden";

      current.style.height = `${current.parentElement.clientHeight}px`;
    }

    return () => {
      current?.parentElement?.style.removeProperty("position");

      current?.parentElement?.style.removeProperty("height");

      current?.parentElement?.style.removeProperty("overflow");
    };
  }, []);

  return (
    <div className={classNames(css.Loading, className)} ref={ref}>
      <Spinner className={css.spinner} type={type} />
    </div>
  );
};

export default Loading;
