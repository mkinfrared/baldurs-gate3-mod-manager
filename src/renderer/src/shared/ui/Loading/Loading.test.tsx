import { render } from "@testing-library/react";

import Loading from "./Loading";

describe("<Loading />", () => {
  const Component = <Loading />;

  it("should be defined", () => {
    expect(Loading).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
