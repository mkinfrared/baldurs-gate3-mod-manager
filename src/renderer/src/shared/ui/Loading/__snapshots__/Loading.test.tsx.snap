// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Loading /> > should match the snapshot 1`] = `
<div
  style="height: 0px; overflow: hidden;"
>
  <div
    class="Loading"
    style="height: 0px;"
  >
    <svg
      class="Spinner spinner"
    >
      oval.svg
    </svg>
  </div>
</div>
`;
